logstash-core/src/main/java/org/logstash/instrument/witness/ErrorWitness.java

```diff
@@ -0,0 +1,100 @@
+package org.logstash.instrument.witness;
+
+import org.logstash.instrument.metrics.gauge.TextGauge;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Witness for errors.
+ */
+public class ErrorWitness {
+
+    private final TextGauge message;
+    private final TextGauge backtrace;
+    private final Snitch snitch;
+
+    public ErrorWitness() {
+        message = new TextGauge("message");
+        backtrace = new TextGauge("backtrace");
+        snitch = new Snitch(this);
+    }
+
+    /**
+     * Stacktrace as a {@link String}
+     *
+     * @param stackTrace The stack trace already formatted for output.
+     */
+    public void backtrace(String stackTrace) {
+        this.backtrace.set(stackTrace);
+    }
+
+    /**
+     * The message of the error.
+     *
+     * @param message human readable error message.
+     */
+    public void message(String message) {
+        this.message.set(message);
+    }
+
+    /**
+     * Get a reference to associated snitch to get discrete metric values.
+     *
+     * @return the associate {@link Snitch}
+     */
+    public Snitch snitch() {
+        return this.snitch;
+    }
+
+    /**
+     * Stacktrace for Java.
+     *
+     * @param throwable The Java {@link Throwable} that contains the stacktrace to output
+     */
+    public void backtrace(Throwable throwable) {
+
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+             PrintStream printStream = new PrintStream(byteArrayOutputStream)) {
+
+            throwable.printStackTrace(printStream);
+            String backtrace = new String(byteArrayOutputStream.toByteArray(), StandardCharsets.UTF_8);
```

Comment:
I will update it for the next review. (or this one if there are other requested changes)